runtime/src/qio/qio_formatted.c

```diff
@@ -1594,13 +1594,21 @@ qioerr qio_channel_print_string(const int threadsafe, qio_channel_t* restrict ch
     if( width < 0 ) width = 0;
 
     if( style->min_width_columns > 0 ) {
-      if( !style->leftjustify && width < style->min_width_columns ) {
+      // right justify
+      if( !style->leftjustify && !style->centjustify && width < style->min_width_columns ) {
         // Put what we need to for getting to the min_width.
         for( i = 0; i < style->min_width_columns - width; i++ ) {
           err = qio_channel_write_char(false, ch, style->pad_char);
           if( err ) goto rewind;
         }
       }
+
+      if( style->centjustify && width < style->min_width_columns ) {
+        for( i = 0; i < (style->min_width_columns - width) / 2; i++ ) {
+          err = qio_channel_write_char(false, ch, style->pad_char);
+          if( err ) goto rewind;
+        }
+      }
```

Comment:
We probably should have a check somewhere that ensures leftjustify and centjustify weren't both set, but I'd expect that to be an internal error rather than a user facing one unless something has gone wrong